diff --git a/node_modules/@sap/cds/libx/_runtime/cds-services/adapter/odata-v4/handlers/read.js b/node_modules/@sap/cds/libx/_runtime/cds-services/adapter/odata-v4/handlers/read.js
index 94e56aa..2323f36 100644
--- a/node_modules/@sap/cds/libx/_runtime/cds-services/adapter/odata-v4/handlers/read.js
+++ b/node_modules/@sap/cds/libx/_runtime/cds-services/adapter/odata-v4/handlers/read.js
@@ -571,7 +571,7 @@ const read = service => {
         await tx.commit(result)
       }
     } catch (e) {
-      err = e
+      err = e.code < 400 ? {...e, code: 422, message: `(${e.code}) ${e.message}`} : e
 
       if (!changeset) {
         // REVISIT: rollback needed if an error occurred before commit attempted -> how to distinguish?
